src/lib/storage/storage_manager.rb

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env ruby
+#
+# encoding: utf-8
+
+# Copyright (c) [2015] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "yastx"
+require "storage"
+
+module Yast
+  module Storage
+    #
+    # Singleton class for the libstorage object.
+    #
+    # You can simply use StorageManager.instance to use it and create a
+    # libstorage instance if there isn't one yet; this will use common default
+    # parameters for creating it.
+    #
+    # If you need special parameters for creating it, you can use
+    # StorageManager.create_instance with a custom ::Storage::Environment
+    # (notice the difference between the global ::Storage namespace which is
+    # libstorage and this Yast::Storage namespace).
+    #
+    # By default (unless disabled in the ::Storage::Environment), creating the
+    # instance will also start hardware probing. This is why there is an alias
+    # name StorageManager.start_probing to make this explicit.
+    #
+    class StorageManager
+      include Yast::Logger
+      #
+      # Class methods
+      #
+      class << self
+        @instance = nil
```

Comment:
Sure I suck at naming things. `Libstorage` is only one (bad) option. I called it `NeedsAGoodName` while writing the code, if you prefer. :wink: But from a practical point of view
```
disks = Libstorage.storage.probed.all_disks
```
is not more complex than the current.
```
disks = StorageManager.instance.probed.all_disks
```

On the other hand, both approaches return a libstorage object, so the external library is completely exposed from the very beginning. You'll need to read the libstorage documentation in order to use the object you are getting, so making the "externality" even more obvious doesn't look like a problem to me.

We could, of course, go even one step further with this approach and do this nasty thing. But I wouldn't recommend it because it would be using the ```Yast::Storage``` module for two things at the same time (namespacing and providing methods).
```ruby
module Yast
  module Storage
    @libstorage_instance = nil
    @libstorage_logger   = nil
        
    # @return [::Storage::Storage] libstorage object
    def self.libstorage_instance
      create_libstorage_instance unless @storage_instance
      @libstorage_instance
    end
        
    # Provide your own Storage::Environment for custom purposes like mocking
    # the hardware probing etc.
    #
    # @return [::Storage::Storage] libstorage object
    def self.create_libstorage_instance(storage_environment: ::Storage::Environment.new(true))
      create_libstorage_logger
      log.info("Creating Storage object")
      @storage = ::Storage::Storage.new(storage_environment)
    end 
      
    private
      
    def self.create_logger
      @logger = StorageLogger.new
      ::Storage.logger = @logger
    end
  end
end
```
Usage would be
```ruby
disk = libstorage_instance.probed.all_disks
```